### PR TITLE
Feature/73963 security draft

### PR DESCRIPTION
**Pull request by jakub-lotko**

### CODE REVIEW

File: src/HE.InvestmentLoans.BusinessLogic.Tests/Security/CommandHandlers/CheckSecurityAnswersTests.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
+using System;
```
Comment (Carq) [inline src/HE.InvestmentLoans.BusinessLogic.Tests/Security/CommandHandlers/CheckSecurityAnswersTests.cs:1]: `CheckSecurityAnswersTests` -> `CheckSecurityAnswersCommandHandler`

File: src/HE.Investments.IntegrationTests/Loans/SecuritySection/Order04DirLoansSubIntegrationTests.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            .UrlEndWith(SecurityPageUrls.CheckYourAnswersSuffix)
+            .HasTitle(SecurityPageTitles.CheckAnswers);
+    }
+
+    [Fact(Skip = LoansConfig.SkipTest)]
+    [Order(2)]
+    public async Task Order02_ShouldDisplayValidationError_WhenNoIsSelectedAndAdditionalInformationIsNotProvided()
+    {
+        // given
+        var dirLoansSubPage = await TestClient.NavigateTo(SecurityPageUrls.DirLoansSub(_applicationId));
```
Comment (Carq) [inline src/HE.Investments.IntegrationTests/Loans/SecuritySection/Order04DirLoansSubIntegrationTests.cs:55]: All tests in class should (Except first test in the class) should take a page from previous step - thanks to that performance of integration tests will be much better

File: src/HE.InvestmentLoans.Common/Domain/ValueObject.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
+namespace HE.InvestmentLoans.Common.Domain;
```
Comment (Carq) [inline src/HE.InvestmentLoans.Common/Domain/ValueObject.cs:1]: Why it is duplication with `HE.InvestmentLoans.Contract.Domain.ValueObject` ?

File: src/HE.InvestmentLoans.BusinessLogic/Security/SecurityEntity.cs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        }
+
+        UncompleteSection();
+    }
+
+    public void ProvideDirectorLoansSubordinate(DirectorLoansSubordinate directLoansSubordinate)
+    {
+        if (directLoansSubordinate.IsNotProvided())
+        {
+            DirectorLoansSubordinate = null!;
```
Comment (Carq) [inline src/HE.InvestmentLoans.BusinessLogic/Security/SecurityEntity.cs:65]: Is it possible?  DirectorLoansSubordinate shouldnt be null.

File: src/HE.InvestmentLoans.BusinessLogic/Security/SecurityEntity.cs [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+            return;
+        }
+
+        if (DirectorLoansDoNotExists())
+        {
+            throw new DomainException("Cannot add director loans subordinate because director loans does not exist.", LoanApplicationErrorCodes.DirectorLoansNotExist);
```
Comment (Carq) [inline src/HE.InvestmentLoans.BusinessLogic/Security/SecurityEntity.cs:72]: Should ` OperationResult.New().AddValidationError` be used?

File: src/HE.InvestmentLoans.WWW/Controllers/SecurityV2Controller.cs [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+using HE.InvestmentLoans.Contract.Security.Queries;
+using HE.InvestmentLoans.Contract.Security.ValueObjects;
+using HE.InvestmentLoans.WWW.Routing;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace HE.InvestmentLoans.WWW.Controllers;
+
+[SuppressMessage("Usage", "CA1801", Justification = "It should be fixed in the future")]
```
Comment (Carq) [inline src/HE.InvestmentLoans.WWW/Controllers/SecurityV2Controller.cs:21]: For new code we should avoid suppress

File: src/HE.InvestmentLoans.WWW/Controllers/SecurityV2Controller.cs [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+[Route("application/{id}/security")]
+[Authorize]
```
Comment (Carq) [inline src/HE.InvestmentLoans.WWW/Controllers/SecurityV2Controller.cs:23]: Authorize -> AuthorizeWithCompletedProfile

File: src/HE.InvestmentLoans.WWW/Controllers/SecurityV2Controller.cs [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+        {
+            ModelState.AddValidationErrors(result);
+
+            var response = await _mediator.Send(new GetSecurity(LoanApplicationId.From(id)), token);
+            return View("ChargesDebtCompany", response.ViewModel);
+        }
+
+        return redirect.IsProvided() ?
+            Change(redirect, new { Id = id }) :
+            await Continue(new { Id = id });
```
Comment (Carq) [inline src/HE.InvestmentLoans.WWW/Controllers/SecurityV2Controller.cs:72]: It would be nice to handle change inside of Continue method

**Comment by Carq** [comment]
Old classes should be deleted:
* `SecurityController.cs`
* `SecurityValidator` 
* Others?

File: src/HE.Investments.IntegrationTests/IntegrationFramework/IntegrationTest.cs [hunk 9]
```diff
@@ -31,10 +31,10 @@ protected T GetSharedData<T>(string key)
         where T : class
     {
 #if DEBUG
-        // if (key == SharedKeys.ApplicationLoanIdInDraftStatusKey)
-        // {
-        //     return ("20a97aa8-6e50-ee11-be6f-002248c652b4" as T)!;
-        // }
+        if (key == SharedKeys.ApplicationLoanIdInDraftStatusKey)
```
Comment (Carq) [inline src/HE.Investments.IntegrationTests/IntegrationFramework/IntegrationTest.cs:34]: Comment this

File: src/HE.Investments.IntegrationTests/Loans/LoansConfig.cs [hunk 10]
```diff
@@ -3,5 +3,5 @@ namespace HE.InvestmentLoans.IntegrationTests.Loans;
 public static class LoansConfig
 {
     // Make it null when you want to run tests locally
-    public const string? SkipTest = "Waits for DevOps configuration - #76791";
+    public const string? SkipTest = null; // "Waits for DevOps configuration - #76791";
```
Comment (Carq) [inline src/HE.Investments.IntegrationTests/Loans/LoansConfig.cs:6]: Please revert

**Review by Carq — APPROVED** [review]

File: src/HE.InvestmentLoans.WWW/Views/Shared/_SecurityAnswers.cshtml [hunk 11]
```diff
@@ -8,24 +8,27 @@
...
+    { "Link", Url.Action("ChargesDebtCompany", "Security", new {id = id,  redirect = "CheckAnswers"}) },
+    { "LinkId", "charges-debt-change"}
     })
 
     @if (Model.ChargesDebtCompany == "Yes")
     {
         @await Html.PartialAsync("_CheckAnswersRow", new ViewDataDictionary(ViewData)
         {
         { "Title", "Debenture" },
-        { "Value",  $"{ Model.ChargesDebtCompany }, { Model.ChargesDebtCompanyInfo }" },
```
Comment (dawid-pawulski) [inline src/HE.InvestmentLoans.WWW/Views/Shared/_SecurityAnswers.cshtml:19]: Undo this value change. It was correct

**Review by dawid-pawulski — APPROVED** [review]

**Pull request merged**